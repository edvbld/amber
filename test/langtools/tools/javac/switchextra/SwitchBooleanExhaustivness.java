/*
 * @test /nodynamiccopyright/
 * @compile/fail/ref=SwitchBooleanExhaustivness.out -XDrawDiagnostics SwitchBooleanExhaustivness.java
 */
public class SwitchBooleanExhaustivness {

    private int exhaustive1(boolean b) {
        switch (b) {
            case false: return 0;
            case true: return 1;
        }
    }

    private int exhaustive2(boolean b) {
        switch (b) {
            case false: return 0;
            default: return 1;
        }
    }

    private int exhaustive3(boolean b) {
        return switch (b) {
            case false: break 0;
            case true: break 1;
        };
    }

    private int exhaustive4(boolean b) {
        return switch (b) {
            case false: break 0;
            default: break 1;
        };
    }

    private int notExhaustive1(boolean b) {
        switch (b) {
            case false: return 0;
        }
    }

    private int notExhaustive2(boolean b) {
        return switch (b) {
            case false: break 0;
        };
    }

}