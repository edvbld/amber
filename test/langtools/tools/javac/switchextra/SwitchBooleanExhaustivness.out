SwitchBooleanExhaustivness.java:39:5: compiler.err.missing.ret.stmt
SwitchBooleanExhaustivness.java:42:16: compiler.err.not.exhaustive
2 errors
